ludwig/features/category_feature.py

```diff
@@ -137,13 +142,36 @@ def get_feature_meta(column, preprocessing_parameters, backend):
 
     @staticmethod
     def feature_data(column, metadata):
-        return column.map(
-            lambda x: (
-                metadata["str2idx"][x.strip()]
-                if x.strip() in metadata["str2idx"]
-                else metadata["str2idx"][UNKNOWN_SYMBOL]
+        def __replace_token_with_idx(value: Any, metadata: Dict[str, Any], unknown_symbol_idx: int) -> int:
+            stripped_value = value.strip()
+            if stripped_value in metadata["str2idx"]:
+                return metadata["str2idx"][stripped_value]
+            logger.warning(
+                f"""
+                Encountered unknown symbol '{stripped_value}' for '{column.name}' during category
+                feature preprocessing. This should never happen during training. If this happens during
+                inference, this may be an indication that not all possible symbols were present in your
+                training set. Consider re-splitting your data to ensure full representation, or setting
+                preprocessing.most_common parameter to be smaller than this feature's total vocabulary
+                size, {len(metadata["str2idx"])}, which will ensure that the model is architected and
+                trained with an UNKNOWN symbol. Returning the index for the most frequent symbol instead.
```

Comment:
We can actually include the most frequent symbol in this warning:

`metadata["idx2str"][fallback_symbol_idx]`